src/constraints/storage_constraints.jl

```diff
@@ -99,6 +106,33 @@ function add_elec_storage_dispatch_constraints(m, p, b; _n="")
             sum(m[Symbol("dvStorageEnergy"*_n)][b])
         )
     end
+
+    if p.s.storage.attr[b] isa ElectricStorage && !isnothing(p.s.storage.attr[b].fixed_duration)
```

Comment:
I don't think we need these isa ElectricStorage checks because we're already inside the function add_elec_storage_dispatch_constraints(m, p, b; _n="") which is only passed ElectricStorage as b